### PR TITLE
[SKT-15876]

### PR DESCRIPTION
**Pull request by jmmklrxscrp**
- added username to log pattern

image

image

### CODE REVIEW

File: src/main/java/com/kjt/logging/Slf4jMDCFilter.java [hunk 1]
```diff
@@ -59,4 +66,18 @@ private void logRequest(HttpServletRequest request) {
         String path = StringUtils.isBlank(queryString) ? uri : uri + "?" + queryString;
         log.info("Request path: {} {}", method, path);
     }
+
+    public static Optional<String> getCurrentUserLogin() {
+        SecurityContext securityContext = SecurityContextHolder.getContext();
+        return Optional.ofNullable(securityContext.getAuthentication())
+                .map(authentication -> {
+                    if (authentication.getPrincipal() instanceof UserDetails) {
```
Comment (mnsuarez) [inline src/main/java/com/kjt/logging/Slf4jMDCFilter.java:74]: Hi @jmmklrxscrp ,

We can refactor this code block into this:
```
return Optional.ofNullable(securityContext.getAuthentication())
                .map(authentication -> {
                    Object principal = authentication.getPrincipal();
                    if (principal instanceof UserDetails) {
                        UserDetails userDetails = (UserDetails) principal;
                        return userDetails.getUsername();
                    } else if (principal instanceof String) {
                        return (String) principal;
                    }
                    return null;
                });
```
Thanks,
Michael

File: src/main/java/com/kjt/logging/Slf4jMDCFilter.java [hunk 2]
```diff
@@ -31,7 +36,9 @@ protected void doFilterInternal(final HttpServletRequest request,
             throws ServletException, IOException {
         try {
             final String correlationId = getCorrelationIdFromHeader(request);
```
Comment (mnsuarez) [inline src/main/java/com/kjt/logging/Slf4jMDCFilter.java:38]: Hi @jmmklrxscrp ,

We can refactor the `getCorrelationIdFromHeader` method to this:
```
private String getCorrelationIdFromHeader(final HttpServletRequest request) {
        String correlationId = request.getHeader(CORRELATION_ID_HEADER_NAME);
        return StringUtils.isBlank(correlationId) ? generateUniqueCorrelationId() : correlationId;
    }
```
Thanks,
Michael

Comment (jmmklrxscrp) [reply src/main/java/com/kjt/logging/Slf4jMDCFilter.java:38]: Changed

**Review by mnsuarez — APPROVED** [review]

**Comment by josephsazon** [comment]
I suggest to put this in log instead along with request path and other info. We only need to log this once per request and use the the correlation ID to identify other logs within the request.

**Comment by jmmklrxscrp** [comment]
I think it is fine for now to put it there. We usually search for the error first, with that search we can already know the user.  In kibana, we can easily map the whole user activity with username.

**Review by josephsazon — APPROVED** [review]

**Pull request merged**